runtime/src/stakes.rs

```diff
@@ -529,15 +561,13 @@ impl PartialEq<StakesEnum> for StakesEnum {
     fn eq(&self, other: &StakesEnum) -> bool {
         match (self, other) {
             (Self::Accounts(stakes), Self::Accounts(other)) => stakes == other,
-            (Self::Accounts(stakes), Self::Delegations(other)) => {
+            (Self::Delegations(stakes), Self::Delegations(other)) => stakes == other,
+            (Self::Stakes(stakes), Self::Stakes(other)) => stakes == other,
+            (stakes, other) => {
                 let stakes = Stakes::<Delegation>::from(stakes.clone());
-                &stakes == other
-            }
-            (Self::Delegations(stakes), Self::Accounts(other)) => {
                 let other = Stakes::<Delegation>::from(other.clone());
-                stakes == &other
+                stakes == other
```

Comment:
It's certainly simpler to read this way, but might be worth avoiding the extra clone (for the cases where one side of the match is a `Self::Delegations`).

Reply:
I was concerned about this too but found out that equality is never checked in perf sensitive areas and was able to put it behind the DCOU feature here: https://github.com/anza-xyz/agave/pull/1395

So I think it's ok to have an extra clone but can switch it back if you prefer.

Reply:
Oh cool, couldn't see that DCOU add on this PR. I'm much less concerned in a test-only context. Thanks.